agent/agent-bootstrap/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/rpc/RpcClientMetrics.java

```diff
@@ -0,0 +1,165 @@
+import static io.opentelemetry.instrumentation.api.instrumenter.rpc.MetricsView.applyClientView;
+import static java.util.logging.Level.FINE;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.instrumentation.api.instrumenter.OperationListener;
+import io.opentelemetry.instrumentation.api.instrumenter.OperationMetrics;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * {@link OperationListener} which keeps track of <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/rpc.md#rpc-client">RPC
+ * client metrics</a>.
+ */
+public final class RpcClientMetrics implements OperationListener {
+
+  private static final double NANOS_PER_MS = TimeUnit.MILLISECONDS.toNanos(1);
+
+  private static final ContextKey<RpcClientMetrics.State> RPC_CLIENT_REQUEST_METRICS_STATE =
+      ContextKey.named("rpc-client-request-metrics-state");
+
+  private static final Logger logger = Logger.getLogger(RpcClientMetrics.class.getName());
+
+  private final DoubleHistogram clientDurationHistogram;
+
+  private RpcClientMetrics(Meter meter) {
+    clientDurationHistogram =
+        meter
+            .histogramBuilder("rpc.client.duration")
+            .setDescription("The duration of an outbound RPC invocation")
+            .setUnit("ms")
+            .build();
+  }
+
+  /**
+   * Returns a {@link OperationMetrics} which can be used to enable recording of {@link
+   * RpcClientMetrics} on an {@link
+   * io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder}.
+   */
+  public static OperationMetrics get() {
+    return RpcClientMetrics::new;
+  }
+
+  @Override
+  public Context onStart(Context context, Attributes startAttributes, long startNanos) {
+    return context.with(
+        RPC_CLIENT_REQUEST_METRICS_STATE,
+        new AutoValue_RpcClientMetrics_State(startAttributes, startNanos));
+  }
+
+  @Override
+  public void onEnd(Context context, Attributes endAttributes, long endNanos) {
+    State state = context.get(RPC_CLIENT_REQUEST_METRICS_STATE);
+    if (state == null) {
+      logger.log(
+          FINE,
+          "No state present when ending context {0}. Cannot record RPC request metrics.",
+          context);
+      return;
+    }
+
+    // START APPLICATION INSIGHTS CODE
+
+    // this is needed for detecting telemetry signals that will trigger pre-aggregated metrics via
+    // auto instrumentations
+    Span.fromContext(context).setAttribute(IS_PRE_AGGREGATED, "True");
```

Comment:
`AttributeKey<Boolean>` is probably clearer here, and then we can map to `"True"` in the exporter